djstripe/admin.py

```diff
@@ -733,3 +740,18 @@ def save_model(self, request, obj: "models.WebhookEndpoint", form, change):
 
                 new_obj = obj.__class__.sync_from_stripe_data(stripe_we)
                 obj.id = new_obj.id
+
+            except stripe.error.InvalidRequestError as error:
+                # todo figure out how to  send the form context back to be redisplayed
+                form.add_error(None, error)
+                raise
+
+    def _changeform_view(
+        self, request, object_id=None, form_url="", extra_context=None
+    ):
+
+        try:
+            return super()._changeform_view(request, object_id, form_url, extra_context)
+        except Exception as error:
+            self.message_user(request, error, level=messages.ERROR)
+            return HttpResponseRedirect(request.path)
```

Comment:
This is definitely not the right way to do it but I like the general idea; I'll take it and run with it.

Reply:
Yes this is a hack because django on purpose doesn't allow raising any error to get captured and get displayed on the admin page. There is an open ticket about this too. So the only way was to raise an exception and catch it in the changelist_view and then redirect user to the same page with the error message displayed.

Reply:
Since we're customizing the form, I think we can do better. I don't think saving of the model on stripe should happen where I put it -- I set it there because that was the easiest way to get started. I can probably move it to the form and from there, catch stripe exceptions and look for which fields should error out.

Reply:
I'm spending some extra time on this model because it'll eventually serve as a template for how we will do things with other admin models in the future.

Reply:
Yes, that would solve all of these issues.

Reply:
Okay, new branch uses _post_clean() and save() on modelform to properly do this.